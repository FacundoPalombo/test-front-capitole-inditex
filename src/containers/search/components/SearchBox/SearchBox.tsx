import React from 'react';
import styles from './styles.module.scss';

import { func, number, string } from 'prop-types';

type SearchBoxProps = {
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  value: string;
  resultsCount: number;
};

const SearchBox = ({ onChange, value, resultsCount = 0 }: SearchBoxProps) => {
  return (
    <span className={styles.search_box}>
      <label
        htmlFor="search_box"
        aria-label="Filter over all the 100 best podcasts on itunes."
        className={styles.search_box__label}
      >
        {resultsCount}
      </label>
      <input
        placeholder="Filter podcasts..."
        id="search_box"
        data-testid="search_box"
        name="search_box"
        type="text"
        className={styles.search_box__input}
        onChange={onChange}
        value={value}
      />
    </span>
  );
};

SearchBox.propTypes = {
  value: string,
  onChange: func,
  resultsCount: number,
};

export default SearchBox;
