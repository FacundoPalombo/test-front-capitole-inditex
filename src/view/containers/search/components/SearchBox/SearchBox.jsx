import React from 'react';
import styles from './styles.module.scss';
import { func, string } from 'prop-types';

const SearchBox = ({ onChange, value }) => {
  return (
    <span className={styles.search_box}>
      <label
        htmlFor="search_box"
        aria-label="Filter over all the 100 best podcasts on itunes."
        className={styles.search_box__label}
      >
        100
      </label>
      <input
        placeholder="Filter podcasts..."
        id="search_box"
        data-testid="search_box"
        name="search_box"
        type="text"
        className={styles.search_box__input}
        onChange={onChange}
        value={value}
      />
    </span>
  );
};

SearchBox.propTypes = {
  value: string,
  onChange: func,
};

export default SearchBox;
