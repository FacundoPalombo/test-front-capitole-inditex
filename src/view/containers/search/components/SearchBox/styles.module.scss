.search_box {
  position: absolute;
  top: 0;
  right: 0;

  &__input {
    color: var(--font-primary);
    padding: 8px 8px 8px 12px;
    margin: 16px 8px 8px 16px;
    border: 1px solid var(--font-disabled);
    border-radius: 4px;
    & ::placeholder {
      font-size: small;
      color: var(--font-disabled);
    }
  }

  &__label {
    background-color: #3976a8;
    padding: 2px 8px;
    font-size: large;
    font-weight: 700;
    color: white;
    border-radius: 5px;
  }
}
