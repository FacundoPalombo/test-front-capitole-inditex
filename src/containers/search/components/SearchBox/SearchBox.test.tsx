import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import SearchBox from './SearchBox';

describe('SearchBox unit tests', () => {
  it('should match snapshot', () => {
    const { asFragment: snap1 } = render(
      <SearchBox onChange={jest.fn()} value="123" resultsCount={2} />
    );
    expect(snap1()).toMatchSnapshot();

    const { asFragment: snap2 } = render(
      <SearchBox onChange={jest.fn()} value="321" resultsCount={2} />
    );
    expect(snap2()).toMatchSnapshot();
  });
  it('should change value when user type', async () => {
    const user = userEvent.setup();
    const onChangeStub = jest.fn();

    const { getByRole } = render(
      <SearchBox onChange={onChangeStub} value="asd" resultsCount={2} />
    );

    const searchInput = getByRole('textbox', {
      name: 'Filter over all the 100 best podcasts on itunes.',
    });
    await user.click(searchInput);
    await user.keyboard('asd');

    expect(searchInput).toHaveValue('asd');
    expect(onChangeStub).toHaveBeenCalledTimes(3);
  });
});
