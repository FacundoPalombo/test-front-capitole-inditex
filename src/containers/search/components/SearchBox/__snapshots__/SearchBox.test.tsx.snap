// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox unit tests should match snapshot 1`] = `
<DocumentFragment>
  <span
    class="search_box"
  >
    <label
      aria-label="Filter over all the 100 best podcasts on itunes."
      class="search_box__label"
      for="search_box"
    >
      2
    </label>
    <input
      class="search_box__input"
      data-testid="search_box"
      id="search_box"
      name="search_box"
      placeholder="Filter podcasts..."
      type="text"
      value="123"
    />
  </span>
</DocumentFragment>
`;

exports[`SearchBox unit tests should match snapshot 2`] = `
<DocumentFragment>
  <span
    class="search_box"
  >
    <label
      aria-label="Filter over all the 100 best podcasts on itunes."
      class="search_box__label"
      for="search_box"
    >
      2
    </label>
    <input
      class="search_box__input"
      data-testid="search_box"
      id="search_box"
      name="search_box"
      placeholder="Filter podcasts..."
      type="text"
      value="321"
    />
  </span>
</DocumentFragment>
`;
